src/app/pages/user-profile/page.jsx

```diff
@@ -1,47 +1,146 @@
 'use client'
 import React, { useState, useEffect } from 'react'
 import { useSession } from 'next-auth/react';
-import { BiSolidUser, BiSolidUserCircle } from 'react-icons/bi';
+import { BiSolidUser, BiSolidUserCircle,BiSolidUserDetail } from 'react-icons/bi';
 import { MdOutlineAlternateEmail } from 'react-icons/md';
 import { AiFillEye, AiFillEyeInvisible } from 'react-icons/ai';
+import { validateEmail, validatePassword, validateTextField, validateCiField } from '@/utils/formValidations';
 import '@/app/pages/user-profile/UserProfile.css'
-
+import api from '@/app/api/api';
 
 const Profile = () => {
   const [showPassword, setShowPassword] = useState(false);
+  const [ci, setCi] = useState('');
   const [name, setName] = useState('');
   const [lastName, setLastName] = useState('');
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
   const [gender, setGender] = useState('default');
+  const [isEditing, setIsEditing] = useState(false);
+  const [user] = useState([])
   const session = useSession()
-  console.log(session)
 
   const togglePasswordVisibility = () => {
     setShowPassword(!showPassword);
   };
+
   const handleInputChange = (e) => {
     const { id, value } = e.target;
-    setName(value);
-    setLastName(value);
-    setEmail(value);
-    setPassword(value);
-  }
+
+    if (id === 'name') {
+      setName(value);
+    } else if (id === 'lastName') {
+      setLastName(value);
+    } else if (id === 'email') {
+      setEmail(value);
+    } else if (id === 'password') {
+      setPassword(value);
+    } else if (id === 'ci') {
+      setCi(value);
+    }
+  };
+
+  const handleEdit = (e) => {
+    e.preventDefault();
+    setIsEditing(true);
+  };
+
+  const handleSave = async (e) => {
+    e.preventDefault();
+    try {
+      const [isEmailValid, emailErrorMessage] = validateEmail(email);
+      const [isPasswordValid, passwordErrorMessage] = validatePassword(password);
+      const [isCiValid, ciErrorMessage] = validateCiField(ci);
+      const [isNameValid, nameErrorMessage] = validateTextField(name, 'Name');
+      const [isLastNameValid, lastNameErrorMessage] = validateTextField(lastName, 'Last Name');
+
+      if (!isEmailValid) {
+        console.error('Email Validation Error:', emailErrorMessage);
+        return;
+      }
+      if (!isPasswordValid) {
+        console.error('Password Validation Error:', passwordErrorMessage);
```

Comment:
Throws user-friendly errors, or changes these errors to throw exceptions such as an `Error`.